runtime/autoload/man.vim

```diff
@@ -190,6 +190,23 @@ function! s:open_page(sect, name)
     silent keepjumps 1delete _
   endwhile
   setlocal filetype=man
+  call man#create_toc()
+endfunction
+
+function! man#create_toc() abort
+  if !exists('b:man_toc')
+    " Cache the TOC since user might switch in and out between the man pages.
+    let b:man_toc = []
+    for lnum in range(1, line('$'))
+      let text = getline(lnum)
+      if text =~# '^\%(\%(\S.*\)\=\S\|\%1l.*\|\s\{3\}\S.*\)$'
```

Comment:
I originally used syntax for finding the lines because I was afraid of syntax changes, but it turned out to be pretty slow on large help files.  ![](http://sae.tweek.us/media/emoticons/emot-argh.gif)

Was there a problem with tabs being matched?